book/src/dev/rfcs/0008-release-planning.md

```diff
@@ -0,0 +1,191 @@
+everywhere.  What should we do outside of Docker?  One solution would be to try
+to create packages for each platform-specific package manager (Homebrew,
+something for Windows, various different flavors of Linux distribution), but
+this creates a large amount of additional work requiring platform-specific
+knowledge.  Worse, this work cannot be outsourced to others without giving up
+control over our software distribution -- if, for instance, a third party
+creates a Homebrew package, and we recommend people install Zebra using that
+package, we're reliant on that third party to continue packaging our software
+forever, or leave our users stranded.
+
+Instead, we can publish `zebrad` as a Rust crate and recommend `cargo install`.
+This approach has two major downsides.  First, installation takes longer,
+because Zebra is compiled locally.  Second, as long as we have a dependency on
+`zcashconsensus`, we'll have to instruct users to install some
+platform-specific equivalent of a `build-essential` package. And as long as
+we depend on `zcash_script`, we'll have to instruct users to install `libclang`.
+However, even for
+crates such as `zcashconsensus` that build native code, the `cargo`-managed
+build process is far, far more reliable than build processes for C or C++
+projects.  We would not be asking users to run autotools or `./configure`, just
+a one-step `cargo install`.  We also know that it's possible to reliably build
+Zebra on each platform with minimal additional steps, because we do so in our
+CI.
+
+In contrast to these downsides, distributing `zebra` through Cargo has a number
+of upsides.  First, because we distribute our libraries using crates.io, we
+already have to manage tooling for publishing to crates.io, so there's no
+additional work required to publish `zebrad` this way.  Second, we get a
+cross-platform update mechanism with no additional work, since `cargo install`
+will upgrade to the latest published version.  Third, we don't rely on any
+third parties to mediate the relationship between us and our users, so users
+can get updates as soon as we publish them.  Fourth, unlike a system package
+manager, we can pin exact hashes of every transitive dependency (via the
+`Cargo.lock`, which `cargo install` can be configured to respect).  Fifth,
+we're positioned to pick up (or contribute to) ecosystem-wide integrity
+improvements like a transparency log for `crates.io` or work on reproducible
+builds for Rust.
+
+This proposal is summarized above in the [guide-level
+explanation](#guide-level-explanation).
+
+## Release Processes
+
+The next question is what kind of release processes and automation we should
+use.  Here are two important priorities for these processes:
+
+1.  Reducing the friction of doing any individual release, allowing us to move
+    closer to a continuous deployment model;
+2.  Reducing the risk of error in the release process.
+
```

Comment:
A potential third priority, from the release discussion:

3. updating should be part of the normal flow of operations for users

If updating is costly and difficult, then updating becomes special mode. New set of failure cases, updating should be normal flow of operations, both non compatible and breaking changes are easier to deal with, how can we make it easy for people to always be updating

Reply:
Looking at rust ecosystem as a whole, even when libraries make breaking changes, the tooling around updates is good enough that in practice software gets updates reasonably effectively

Reply:
Breaking changes are easier to deal with if more frequent and clearly scoped, rather than one big pile of unrelated changes.

For example: config changes.

We can also make breaking changes opt-in, but that creates an extra mode, which has maintenance costs.

Another strategy: make a breaking change, then go and fix all dependencies, works well in more structured environment like crates io with package registry.

For example, crater (a tool in rust ecosystem) allows rustlang developers to check the impact of API stability changes.